src/cudadev/DataFormats/SoAView.h

```diff
@@ -0,0 +1,375 @@
+                                                                                                                                    \
+  /* Constructor relying on user provided stores */                                                                                 \
+  SOA_HOST_ONLY CLASS ( _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_CONSTRUCTION_PARAMETERS, const, STORES_LIST) ):                         \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_MEMBER_INITIALIZERS, ~, VALUE_LIST) {}                                                    \
+                                                                                                                                    \
+  struct const_element {                                                                                                            \
+    SOA_HOST_DEVICE_INLINE                                                                                                          \
+    const_element(size_t index,                                                                                                     \
+      /* Declare parameters */                                                                                                      \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_ELEMENT_VALUE_ARG, const, VALUE_LIST)                                                     \
+    ):                                                                                                                              \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_CONST_ELEM_MEMBER_INIT, index, VALUE_LIST)                                                \
+       {}                                                                                                                           \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_CONST_ELEMENT_ACCESSOR, ~, VALUE_LIST)                                                            \
+                                                                                                                                    \
+  private:                                                                                                                          \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_CONST_ELEMENT_VALUE_MEMBER, ~, VALUE_LIST)                                                        \
+  };                                                                                                                                \
+                                                                                                                                    \
+  /* AoS-like accessor (const) */                                                                                                   \
+  SOA_HOST_DEVICE_INLINE                                                                                                            \
+  const const_element operator[](size_t index) const {                                                                              \
```

Comment:
Confusing `const`
```suggestion
  const_element operator[](size_t index) const {                                                                              \
```

Reply:
This redudancy was removed for views (and gone for layouts, as they do not have accessors).